sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/RapidsHiveErrors.scala

```diff
@@ -54,7 +54,7 @@ object RapidsHiveErrors {
 
   def cannotResolveAttributeError(name: String, outputStr: String): Throwable = {
     new AnalysisException(
-      s"Unable to resolve $name given [$outputStr]")
+      s"Unable to resolve $name given [$outputStr]") {}
```

Comment:
I don't think this is a good approach, as it's very unfriendly to the end-user.  The exception is going to show up without AnalysisException in the name -- it will be something like RapidsHiveErrors$$anon$## (or whatever the enclosing class is where the anonymous class was created), and thus won't look at all like the AnalysisException class that the user is used to seeing.  Yes, it's semantically like an AnalysisException since it derives from it, but the way it appears to the end user in a stacktrace does not look similar at all.

The reason Spark made the string private is because they want the code to go through QueryExecutionErrors and other similar error reporting interfaces.  We have our own wrappers to that in shims, so maybe the best way to fix this is to migrate these use cases to those already shim'd error interfaces, and have the older shims call AnalysisException and the new shims use the newer error reporting that doesn't use an arbitrary string.  If we really want to match the Spark behavior here, that's the right way to do it.

Reply:
I agree, that's the right way to do it and went down that path but I wasn't sure if we will be OK with the difficulty in code review and decided to post this before going back that route. Will update in a few